core/build/src/lib.rs

```diff
@@ -7,6 +7,7 @@ use bindgen::{
     callbacks::{IntKind, ParseCallbacks},
     Builder, EnumVariation,
 };
+use std::collections::HashSet;
```

Comment:
It isn't needed, no. It was something leftover from development, but I thought it makes sense here. There is never a case where we want repeat derives on the same type because it is always a build error. By using a set instead of a vector, we can make sure that never happens. It might let us take shortcuts with the logic in this function in the future. By allowing us to add duplicate derives, we can handle more cases with fewer branches.

That said, I'm not against removing it. If we want to keep it though, there are a few other vectors in this file that I think should also be replaced.

Reply:
I think it's fine to leave as is. The argument of "it is always an error" to have a derive show up more than once is good justification for it. I wouldn't worry about updating the others, unless/until we see an issue.